### ISSUE TITLE
Detach / Stop Gradients?

### ISSUE THREAD

**Issue by oswinso** [casadi/casadi#2928]
Is there any way to "detach" or "stop" gradients to treat an expression as "constant" in a computation? ex.

```python
x = SX.sym("x", 2)
max_x = mmax(x).detach()
unnormalized = exp(x - max_x)
softmax = unnormalized / sum1(unnormalized)
```
